Modules/mmapmodule.c

```diff
@@ -682,6 +682,80 @@ mmap__exit__method(PyObject *self, PyObject *args)
     return _PyObject_CallMethodId(self, &PyId_close, NULL);
 }
 
+static PyObject *
+mmap__repr__method(PyObject *self)
+{
+    PyObject *repr;
+    mmap_object *m_obj;
+    m_obj = (mmap_object *) self;
+
+    const char *reprfmt;
+    const char *access_str;
+
+    switch (m_obj->access) {
+        case ACCESS_DEFAULT:
+            access_str = "ACCESS_DEFAULT";
+            break;
+
+        case ACCESS_READ:
+            access_str = "ACCESS_READ";
+            break;
+
+        case ACCESS_WRITE:
+            access_str = "ACCESS_WRITE";
+            break;
+
+        case ACCESS_COPY:
+            access_str = "ACCESS_COPY";
+            break;
+
+        default:
+            // should not get here
+            assert(0);
+            return NULL;
+    }
+
+    Py_ssize_t size = m_obj->size;
+    const char *tp_name = Py_TYPE(self)->tp_name;
+
+
+#ifdef MS_WINDOWS
+    int fileno = m_obj->file_handle;
+    if (m_obj->map_handle == NULL)
+#elif defined(UNIX)
+    int fileno = m_obj->fd;
+    if (m_obj->data == NULL)
+#endif
+    {
+        reprfmt = "<%s closed=True fileno=%d access=%s>";
+        repr = PyUnicode_FromFormat(reprfmt, tp_name, fileno, access_str);
+    }
+    else {
+        const char *data = m_obj->data;
+        Py_ssize_t pos = m_obj->pos;
+
+        PyObject *length = PyLong_FromSize_t(size);
+        PyObject *offset = PyLong_FromSize_t(pos);
```

Comment:
C APIs could fail. You need to check here and if they fail, let the function returns and propagates exceptions.

Reply:
Yeah we don't really need this. I used `PyObject` for arbitrary precision, but it seems that `mmap_object` supports only long integer size. Thus I use `%lu` instead.

Reply:
`pos` and `size` are both of type `Py_ssize_t`, you don't need to care what actually precision it is for represention case, the format function will take care of it. `%lu` is not the exact format here, checkout https://docs.python.org/3/c-api/unicode.html#c.PyUnicode_FromFormat.

Reply:
Okay, thanks for this!